pkg/serviceaccounts/oauthclient/oauthclientregistry.go

```diff
@@ -459,9 +473,29 @@ func (a *saOAuthClientAdapter) getServiceAccountTokens(sa *corev1.ServiceAccount
 	tokens := []string{}
 	for i := range allSecrets.Items {
 		secret := &allSecrets.Items[i]
-		if serviceaccount.IsServiceAccountToken(secret, sa) {
-			tokens = append(tokens, string(secret.Data[kapi.ServiceAccountTokenKey]))
+		if IsServiceAccountToken(secret, sa) {
+			tokens = append(tokens, string(secret.Data[corev1.ServiceAccountTokenKey]))
 		}
 	}
 	return tokens, nil
 }
+
+// IsServiceAccountToken returns true if the secret is a valid api token for the service account
+func IsServiceAccountToken(secret *corev1.Secret, sa *corev1.ServiceAccount) bool {
```

Comment:
Is there a reason why we need to make this function public ?

Reply:
> Is there a reason why we need to make this function public ?

Not at the present time.  It is a copy from upstream kube/kube.